impl/maven-core/src/test/java/org/apache/maven/model/WrapperPropertiesTest.java

```diff
@@ -0,0 +1,264 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.model;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class WrapperPropertiesTest {
+    private Map<String, String> backingStore;
+    private AtomicBoolean getterCalled;
+    private AtomicBoolean setterCalled;
+    private WrapperProperties wrapper;
+
+    @BeforeEach
+    void setUp() {
+        backingStore = new HashMap<>();
+        getterCalled = new AtomicBoolean(false);
+        setterCalled = new AtomicBoolean(false);
+
+        Supplier<Map<String, String>> getter = () -> {
+            getterCalled.set(true);
+            return new HashMap<>(backingStore);
+        };
+
+        Consumer<Properties> setter = props -> {
+            setterCalled.set(true);
+            backingStore.clear();
+            props.forEach((k, v) -> backingStore.put(k.toString(), v.toString()));
+        };
+
+        wrapper = new WrapperProperties(getter, setter);
+    }
+
+    @Test
+    void testInitialization() {
+        wrapper.getProperty("any");
+        assertTrue(getterCalled.get());
+    }
+
+    @Test
+    void testBasicOperations() {
+        // Test set and get
+        wrapper.setProperty("key1", "value1");
+        assertEquals("value1", wrapper.getProperty("key1"));
+        assertEquals("value1", backingStore.get("key1"));
+        assertTrue(setterCalled.get(), "Setter should be called after modification");
+
+        // Test remove
+        wrapper.remove("key1");
+        assertNull(wrapper.getProperty("key1"));
+        assertFalse(backingStore.containsKey("key1"));
+
+        // Test contains
```

Comment:
this should be a separate method

Reply:
idk thats part of the test. but yes dedicated each concern to test `one thing only` as well.